homeassistant/components/homeassistant_sky_connect/__init__.py

```diff
@@ -24,19 +75,28 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         # The USB dongle is not plugged in
         raise ConfigEntryNotReady
 
-    usb_info = usb.UsbServiceInfo(
-        device=entry.data["device"],
-        vid=entry.data["vid"],
-        pid=entry.data["pid"],
-        serial_number=entry.data["serial_number"],
-        manufacturer=entry.data["manufacturer"],
-        description=entry.data["description"],
-    )
+    addon_info = await _multi_pan_addon_info(hass, entry)
+
+    if not addon_info:
```

Comment:
This flow will always assumes that if the add-on is installed, even if it is not running, that we want to use it. 

That means that the user installing the add-on and then restarting Home Assistant, will end up with using the add-on.

I don't think that the add-on being installed should be the indicator how the SkyConnect should be used. Instead, we should rely on data stored in the config entry.

Reply:
No, additional data should not be added to the SkyConnect config entry.
If the Multi PAN addon is configured to use the SkyConnect, we should not discover a ZHA USB flow which will only interfere with the Multi PAN addon.

Reply:
well the question is, what do we want the source of truth to be and what are the pros and cons of storing the user preference in skyconnect config entry vs ZHA configured port vs if the add-on is installed?

Reply:
For now, I think this stateless approach will work for any common setup. Once we get firmware swapping working I think it can be revisited.

Storing firmware state in the config entry would help switch back to EmberZNet firmware, since keeping the SkyConnect integration stateless means we can't tell what's running on the stick until we observe ZHA running/the addon successfully start with firmware flashing enabled. On the other hand, it would not work when the SkyConnect was flashed externally and then plugged back in.

Honestly, I feel automating too much will make any solution fragile. Maybe we can un-hide the "SkyConnect" and "Yellow" integrations and just have a dumb options flow in each to blindly flash one of the two usable firmwares, with a backup manual file upload option? The firmware flasher _is_ asyncio-native and can be run directly from HA Core without an addon...